src/svgp.jl

```diff
@@ -17,55 +20,60 @@ which can be found in closed form.
 variational Gaussian process classification." Artificial Intelligence and
 Statistics. PMLR, 2015.
 """
-function AbstractGPs.approx_posterior(::SVGP, fz::FiniteGP, q::AbstractMvNormal)
+function AbstractGPs.posterior(svgp::SVGP)
+    q, fz = svgp.q, svgp.fz
     m, A = mean(q), _chol_cov(q)
     Kuu = _chol_cov(fz)
     B = Kuu.L \ A.L
     α = Kuu \ (m - mean(fz))
-    data = (A=A, m=m, Kuu=Kuu, B=B, α=α, u=fz.x)
-    return ApproxPosteriorGP(SVGP(), fz.f, data)
+    data = (A=A, m=m, Kuu=Kuu, B=B, α=α)
+    return ApproxPosteriorGP(svgp, fz.f, data)
 end
 
-function Statistics.mean(f::ApproxPosteriorGP{SVGP}, x::AbstractVector)
-    return mean(f.prior, x) + cov(f.prior, x, f.data.u) * f.data.α
+AbstractGPs.posterior(svgp::SVGP, ::FiniteGP, ::AbstractVector) = posterior(svgp)
```

Comment:
Should this not check that e.g. the GP in the second argument is the same as in the first ?

Reply:
It probably should, but it seemed a bit unnecessary to check for that when `fx` is just thrown away?

Reply:
It could catch bugs/user errors when you mean to do one thing but pass the wrong things in, so I think it'd be helpful to have. You can only throw them away *because* you expect them to be the same...